src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,22 +1,27 @@
-import {Component} from '@angular/core';
-import {productsMock} from '../../../shared/products/products.mock';
+import {ChangeDetectionStrategy, Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from '../../../shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CardComponent {
-    readonly product = productsMock[0];
+    @Input() product: Product | null = null;
+
+    @Output() readonly productClick = new EventEmitter<Product['_id']>();
 
     onProductBuy(event: Event) {
         event.stopPropagation();
 
         // eslint-disable-next-line no-console
         console.log('Buy product');
+        //
+        this.productClick.emit(this.product?._id);
```

Comment:
Смотри, `?.` на самом деле может вернуть `undefined` за счет его внутренней реализации. По это сейчас в `EventEmitter` у тебя может передасться не только `Product['_id']`, но и `undefined` из-за `?.`. По этому я бы рекомендовал данную проверку:
```suggestion
        if (this.product) {
                this.productClick.emit(this.product?._id);
        }
```

Так не придется править типизацию и все будет надежно и правдиво

Reply:
поняла, поправила!
спасибо!